src/Internal/BalanceTx/Collateral/Select.purs

```diff
@@ -158,16 +159,27 @@ asTxUnspentOutput
   :: TransactionInput /\ TransactionOutput -> TransactionUnspentOutput
 asTxUnspentOutput (input /\ output) = wrap { input, output }
 
-adaValue :: TransactionUnspentOutput -> BigInt
+adaValue :: TransactionUnspentOutput -> Coin
 adaValue =
-  Value.valueToCoin' <<< _.amount <<< unwrap <<< _.output <<< unwrap
-
-consumeUtxoAdaValue :: BigInt -> TransactionUnspentOutput -> BigInt
-consumeUtxoAdaValue acc = add acc <<< adaValue
-
-nonAdaAsset :: TransactionUnspentOutput -> NonAdaAsset
+  Value.getCoin <<< _.amount <<< unwrap <<< _.output <<< unwrap
+
+consumeUtxoAdaValue :: Coin -> TransactionUnspentOutput -> Coin
+consumeUtxoAdaValue acc = unsafeFromMaybe "consumeUtxoAdaValue" <<< Coin.add acc
+  <<< adaValue
+
+unsafeFromMaybe :: forall a. String -> Maybe a -> a
+unsafeFromMaybe e a = case a of
+  Nothing ->
+    unsafePerformEffect $ throw $ "unsafeFromMaybe: impossible happened: "
+      <> e
+      <> " (please report as bug at "
+      <> bugTrackerLink
+      <> " )"
+  Just v -> v
```

Comment:
I think we should extract this function to Utils / Helpers module and use it in place of uninformative `fromJust` across the entire codebase